providers/util.go

```diff
@@ -29,3 +29,20 @@ func makeOIDCHeader(accessToken string) http.Header {
 	}
 	return makeAuthorizationHeader(tokenTypeBearer, accessToken, extraHeaders)
 }
+
+// Return true if secondArray is a subset of firstArray
+func isSubarray(firstArray, secondArray []string) bool {
+	arraySet := make(map[string]bool)
```

Comment:
Nit, better to use `map[string]struct{}` and then `arraySet[value] = struct{}{}` when setting it and `if _, found := arraySet[value]; !found` when checking, this is more explicit and idiomatic in Go.

Reply:
I truly appreciate the tip! I changed it now.